client/packages/common/src/ui/forms/JsonForms/components/DateTime.tsx

```diff
@@ -63,22 +73,39 @@ const UIComponent = (props: ControlProps) => {
         <FormLabel sx={{ fontWeight: 'bold' }}>{label}:</FormLabel>
       </Box>
       <Box flexBasis={FORM_INPUT_COLUMN_WIDTH}>
-        <BaseDateTimePickerInput
-          // undefined is displayed as "now" and null as unset
-          value={data ?? null}
-          onChange={e => {
-            try {
-              setError('');
-              if (e) handleChange(path, e.toISOString());
-            } catch (err) {
-              setError((err as Error).message);
-              console.error(err);
-            }
-          }}
-          inputFormat="dd/MM/yyyy hh:mm"
-          error={error || props.errors}
-          // readOnly={!!props.uischema.options?.['readonly']}
-        />
+        {!dateOnly ? (
+          <BaseDateTimePickerInput
+            // undefined is displayed as "now" and null as unset
+            value={data ?? null}
+            onChange={e => {
+              try {
+                setError('');
+                if (e) handleChange(path, e.toISOString());
+              } catch (err) {
+                setError((err as Error).message);
+                console.error(err);
+              }
+            }}
+            inputFormat="dd/MM/yyyy hh:mm"
+            error={error || props.errors}
+            // readOnly={!!props.uischema.options?.['readonly']}
+          />
+        ) : (
+          <BaseDatePickerInput
+            // undefined is displayed as "now" and null as unset
+            value={data ?? null}
+            onChange={e => {
+              if (e)
+                handleChange(
+                  path,
+                  // By default, will set current date-time. However, if a date
+                  // is selected, the time will be considered "midnight"
+                  dateFormatter(e, 'yyyy-MM-dd') + ' 00:00:00'
```

Comment:
There is this error try catch for the other case, do we need it here as well? (why do we need it above?)

Is it possible to have dateFormatter and inputFormat calculated beforehand and then handle both cases in one <BaseDatePickInput > block?

Reply:
Yeah, okay, I've refactored all the common code into a single block before the render block: 94feba43b